### PR TITLE
[EN DateTimeV2] Add support for time range with duration

### PR DESCRIPTION
**Pull request by MichaelMWW**
Add support for time range with duration, supported patterns as below.
- from 6 am for 2 hours
- for 2 hours from 6 am
- 16th Oct from 6 am for 2 hours
- 16th Oct for 2 hours from 6 am

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimePeriodExtractor.cs [hunk 1]
```diff
@@ -153,6 +154,22 @@ private List<Token> MatchSimpleCases(string text)
             return ret;
         }
 
+        // Cases like "from 6am for 3 hours" and "for 3 hours from 6 am" are extracted as timerange here.
+        private List<Token> MatchTimePeriodWithDurationCases(string text)
```
Comment (aurghob) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimePeriodExtractor.cs:158]: Is this code being used by an english implementation class ?

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs [hunk 2]
```diff
@@ -476,6 +477,12 @@ private DateTimeResolutionResult ParseBaseDatePeriod(string text, DateObject ref
                 innerResult = ParseDatePointWithAgoAndLater(text, referenceDate);
             }
 
+            // Cases like "for x weeks/days from today/12 sep etc."
+            if (!innerResult.Success)
+            {
+                innerResult = ParseDatePointWithForPrefix(text, referenceDate);
```
Comment (aurghob) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs:483]: Can this be moved to the english implementation if it exists ?

**Review by aurghob — APPROVED** [review]

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimePeriodExtractor.cs (hunk 1 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimePeriodExtractor.cs:158]: As we talked, will consider it in next sprint.

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs (hunk 2 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs:483]: As we talked, will consider it in next sprint.

**Pull request merged**